### ISSUE TITLE
🐛 ProcessEngine breaks when changing the Participant Id of a deployed Diagram

### ISSUE THREAD

**Issue by robpal94** [process-engine/process_engine_runtime#202]
#### Describe your issue

When changing the Participant ID of a diagram that is already deployed, the ProcessEngine cannot be accessed via BPMN Studio. 

Even after restarting the ProcessEngine Runtime, the ProcessEngine cannot be reached from BPMN - Studio. You have to drop your current Database to make the ProcessEngine reachable from BPMN Studio again.

There is also no error thrown on the ProcessEngine site.

#### How can others reproduce the issue?

* Deploy a Process
* Change the PoolId
* Deploy the Process with the changed PoolId. 

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

**Comment by ElRaptorus** [comment]
@robpal94 Is there any log output at all when you attempt to start the runtime?

**Comment by robpal94** [comment]
I cant find any errors in the logs. See: 
[output.txt](https://github.com/process-engine/process_engine_runtime/files/2746098/output.txt)

**Comment by ElRaptorus** [comment]
@robpal94 
Please paste it as a snippet next time for easy reference.

```sh

> @process-engine/process_engine_runtime@5.1.0 start /Users/robin/Documents/process_engine/process_engine_runtime
> node ./bin/index.js

2019-01-10T15:58:42.393Z - info: [processengine:runtime:startup] Running migrations...
2019-01-10T15:58:42.626Z - info: [essential-projects:sequelize_connection_manager] Connection to database '/Users/robin/Library/Application Support/process_engine_runtime/databases/correlation.sqlite' established.
2019-01-10T15:58:42.669Z - info: [essential-projects:sequelize_connection_manager] Disposing connection to '/Users/robin/Library/Application Support/process_engine_runtime/databases/correlation.sqlite'...
2019-01-10T15:58:42.669Z - info: [essential-projects:sequelize_connection_manager] Done.
2019-01-10T15:58:42.670Z - info: [essential-projects:sequelize_connection_manager] Connection to database '/Users/robin/Library/Application Support/process_engine_runtime/databases/external_task.sqlite' established.
2019-01-10T15:58:42.675Z - info: [essential-projects:sequelize_connection_manager] Disposing connection to '/Users/robin/Library/Application Support/process_engine_runtime/databases/external_task.sqlite'...
2019-01-10T15:58:42.675Z - info: [essential-projects:sequelize_connection_manager] Done.
2019-01-10T15:58:42.675Z - info: [essential-projects:sequelize_connection_manager] Connection to database '/Users/robin/Library/Application Support/process_engine_runtime/databases/flow_node_instance.sqlite' established.
2019-01-10T15:58:42.680Z - info: [essential-projects:sequelize_connection_manager] Disposing connection to '/Users/robin/Library/Application Support/process_engine_runtime/databases/flow_node_instance.sqlite'...
2019-01-10T15:58:42.680Z - info: [essential-projects:sequelize_connection_manager] Done.
2019-01-10T15:58:42.680Z - info: [essential-projects:sequelize_connection_manager] Connection to database '/Users/robin/Library/Application Support/process_engine_runtime/databases/process_model.sqlite' established.
2019-01-10T15:58:42.684Z - info: [essential-projects:sequelize_connection_manager] Disposing connection to '/Users/robin/Library/Application Support/process_engine_runtime/databases/process_model.sqlite'...
2019-01-10T15:58:42.684Z - info: [essential-projects:sequelize_connection_manager] Done.
2019-01-10T15:58:42.684Z - info: [processengine:runtime:startup] Migrations successfully executed.
2019-01-10T15:58:44.489Z - verbose: [processengine:persistence:flow_node_instance_repository] Initializing Sequelize connection and loading models...
2019-01-10T15:58:44.490Z - info: [essential-projects:sequelize_connection_manager] Connection to database '/Users/robin/Library/Application Support/process_engine_runtime/databases/flow_node_instance.sqlite' established.
2019-01-10T15:58:44.502Z - verbose: [processengine:persistence:flow_node_instance_repository] Done.
2019-01-10T15:58:44.503Z - verbose: [processengine:persistence:correlation_repository] Initializing Sequelize connection and loading models...
2019-01-10T15:58:44.503Z - info: [essential-projects:sequelize_connection_manager] Connection to database '/Users/robin/Library/Application Support/process_engine_runtime/databases/correlation.sqlite' established.
2019-01-10T15:58:44.506Z - verbose: [processengine:persistence:correlation_repository] Done.
2019-01-10T15:58:44.506Z - verbose: [processengine:persistence:process_definition_repository] Initializing Sequelize connection and loading models...
2019-01-10T15:58:44.506Z - info: [essential-projects:sequelize_connection_manager] Connection to database '/Users/robin/Library/Application Support/process_engine_runtime/databases/process_model.sqlite' established.
2019-01-10T15:58:44.509Z - verbose: [processengine:persistence:process_definition_repository] Done.
2019-01-10T15:58:44.517Z - verbose: [processengine:persistence:external_task_repository] Initializing Sequelize connection and loading models...
2019-01-10T15:58:44.517Z - info: [essential-projects:sequelize_connection_manager] Connection to database '/Users/robin/Library/Application Support/process_engine_runtime/databases/external_task.sqlite' established.
2019-01-10T15:58:44.520Z - verbose: [processengine:persistence:external_task_repository] Done.
2019-01-10T15:58:44.539Z - info: [processengine:runtime:startup] Bootstrapper started successfully.
2019-01-10T15:58:44.539Z - info: [processengine:runtime:startup] Resuming previously interrupted ProcessInstances...
2019-01-10T15:58:44.541Z - info: [processengine:runtime:resume_process_service] Resuming ProcessInstances that were not yet finished.
2019-01-10T15:58:44.548Z - verbose: [processengine:runtime:resume_process_service] Found 0 ProcessInstances to resume.
2019-01-10T15:58:44.548Z - info: [processengine:runtime:startup] Done.
```

From what the log shows, the ProcessEngine is up and running
See here:
```sh
2019-01-10T15:58:44.539Z - info: [processengine:runtime:startup] Bootstrapper started successfully.
```

Have you tried accessing any routes? 
What errors do you get?

**Comment by ElRaptorus** [comment]
@robpal94 
The error occurs because of the restriction that a ProcessDefinition must use the same name as  the ProcessModel it contains.
When deleting a participant and adding a new one, a new ProcessModelId is generated, which the BPMN Studio prevents the user from editing.

If you now deploy that process, the ProcessDefinition name no longer matches the ProcessModel name, so trying to retrieve the ProcessModelList now fails.
Since that function is executed regularly, the ProcessEngine *appears* to be offline, when in fact it isn't really.

This issue will most likely be solved, when #125 is done.

/cc @Vyperus @alexanderkasten @SteffenKn @roblen45 @cmg-dev

**Comment by ElRaptorus** [comment]
Will be treated exclusively as a part of #125 from here on in.
This is something we will be working on when implementing the AtlasEngine.
